client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -305,6 +305,22 @@ Undocumented.prototype.getJetpackConnectionStatus = function( siteId, fn ) {
 	}, fn );
 };
 
+/*
+ * Retrieve current user's connection data for a Jetpack site.
+ *
+ * @param {int}      [siteId]
+ * @param {Function} fn
+ * @api public
+ */
+Undocumented.prototype.getJetpackUserConnectionData = function( siteId, fn ) {
```

Comment:
Yes, the `isRequesting` logic is just redundant (in contrast to how the data layer handles tha tfor us) when used to check if a request is already being done from inside a Query component, but we're trusting in this flag for UI matters all along settings too.